Count cop

rrosenblum opened the pull request:
New performance cop that I thought of the other day when I caught myself making this mistake.

bbatsov commented on lib/rubocop/cop/performance/count.rb:
```diff
@@ -0,0 +1,59 @@
+      #
+      # @example
+      #   # bad
+      #   [1, 2, 3].select { |e| e > 2 }.size
```
You should also account for `length`.

rrosenblum commented:
I have updated the cop to account for `length`

bbatsov commented:
You'll also need to rebase this.

rrosenblum commented:
You are right, I added a number of things to the commit to fix offenses that should have been in the commit for the new cop. I have moved the changes around, updated the commit message, and rebased the code.

rrosenblum commented:
The code has been updated and rebased.
